pkg/util/httputil/http.go

```diff
@@ -4,24 +4,43 @@
 package httputil
 
 import (
+	"crypto/tls"
 	"fmt"
 	"io"
 	"net/http"
+	"time"
 
 	"github.com/cenkalti/backoff/v4"
 	"github.com/pingcap/tiproxy/lib/util/errors"
 	pnet "github.com/pingcap/tiproxy/pkg/proxy/net"
 )
 
-func Get(httpCli http.Client, addr, path string, b backoff.BackOff) ([]byte, error) {
+type Client struct {
+	cli          *http.Client
+	getTLSConfig func() *tls.Config
+}
+
+func NewHTTPClient(getTLSConfig func() *tls.Config) *Client {
+	return &Client{
+		cli:          http.DefaultClient,
+		getTLSConfig: getTLSConfig,
+	}
+}
+
+func (c *Client) SetTimeout(t time.Duration) {
+	c.cli.Timeout = t
+}
+
+func Get(httpCli Client, addr, path string, b backoff.BackOff) ([]byte, error) {
 	schema := "http"
-	if v, ok := httpCli.Transport.(*http.Transport); ok && v != nil && v.TLSClientConfig != nil {
+	if tlsConfig := httpCli.getTLSConfig(); tlsConfig != nil {
 		schema = "https"
+		httpCli.cli.Transport = &http.Transport{TLSClientConfig: tlsConfig}
```

Comment:
It's fine to initialize it in `NewHTTPClient`. `tlsConfig` updates itself after reloading config.